editor/scene_tree_dock.cpp

```diff
@@ -319,6 +319,12 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 				editor->push_item(existing.ptr());
 			else {
 				String path = selected->get_filename();
+				if (path == "")
+				{
+					String root_path = editor_data->get_edited_scene_root()->get_filename();
+					if (root_path == "") path = "res://" + selected->get_name();
```

Comment:
Please don't use one-line ifs. It makes it hard to see the condition and statement.